### PR TITLE
fix: gateway explorer links

### LINKED ISSUE #242: "view on mempool.space" and "view on amboss" don't work on mutinynet
"view on mempool.space" should point to mutinynet.com. Since I don't think there is a node explorer on mutinynet, I think we should just hide the "view on amboss" button.

image

### PR DESCRIPTION
**Pull request by okjodom**
When gateway is deployed on mainnet and testnet, gateway address and node links should point to valid mempool.space links.
On signet, we assume gateway is deployed on mutinynet.com where we point both links. When gateway is on regtest, or the network is not yet known, we don't show "view in explorer" links

Changes in gateway UI take effect after https://github.com/fedimint/fedimint/pull/3455 which introduces `network` field to gateway info. To validate these changes before the upstream changes modify the `network` prop in `FederationCard` and  observe how the links are affected. Valid network values are `"main" | "test" | "signet" | "regtest"`

closes #242

### CODE REVIEW

File: apps/gateway-ui/src/languages/es.json
Comment (EthnTuttle) [inline apps/gateway-ui/src/languages/es.json:1]: Does this break or does it default to English?

Comment (okjodom) [reply apps/gateway-ui/src/languages/es.json:1]: defaults to english since the translation key is renamed

**Review by EthnTuttle — DISMISSED** [review]
LGTM.
Pending https://github.com/fedimint/fedimint/pull/3455, will merge.

**Comment by okjodom** [comment]
> LGTM. Pending [fedimint/fedimint#3455](https://github.com/fedimint/fedimint/pull/3455), will merge.

Thought to clarify that this change hides the link when the `network` prop is undefined. IMO makes it okay to merge before upstream lands. When upstream lands and gateways update, the link will be shown per the network configured

File: apps/gateway-ui/src/languages/en.json
Comment (wbobeirne) [inline apps/gateway-ui/src/languages/en.json:72]: Not important, but we could change this to `View on {{origin}}` and update the `t` calls to include the origin of the link they pulled together. Not really worth holding up the PR, but ideally we avoid deviating from design wherever possible.

File: apps/gateway-ui/src/components/DepositCard.tsx
Comment (wbobeirne) [inline apps/gateway-ui/src/components/DepositCard.tsx:109]: Backwards compatibility could be maintained with always rendering this, and `default`ing to the mainnet link in the switch statement.

**Review by wbobeirne — COMMENTED** [review]
As a point of policy, I'm also hesitant to merge changes that remove functionality with the current `master` of fedimint. I think when we want to target future changes, the game plan should be to maintain backwards compatibility (i.e. default to the mainnet link when `!network`) but priming the app to be ready for an incoming change when the API updates to get the new response type makes the most sense.

If you've got the time to make it backwards compatible, it'd be good to get into the habit of doing that. Otherwise we could just wait for the fedimint PR to land, there's no value in getting this into a `0.1.x` release unless a gateway release is also ported with this network property, which I suspect won't happen.

File: apps/gateway-ui/src/languages/en.json
Comment (okjodom) [reply apps/gateway-ui/src/languages/en.json]: Interpolation with origin results in this madness `"View on https:&#x2F;&#x2F;mempool.space"`. Went with `View on {{host}}`

**Comment by okjodom** [comment]
> If you've got the time to make it backwards compatible, it'd be good to get into the habit of doing that. Otherwise we could just wait for the fedimint PR to land, there's no value in getting this into a 0.1.x release unless a gateway release is also ported with this network property, which I suspect won't happen.

Updated the change to make it backward compatible in UI as a start

**Review by wbobeirne — APPROVED** [review]

**Pull request merged**
